### ISSUE TITLE
River of Flame

### ISSUE THREAD

**Issue by Birk14** [kolton/d2bot-with-kolbot#731]
Sorry if this is the wrong place to ask, but is there a way to get the bot to clear river of flame on the way to chaos sanc and diablo?

**Comment by DarkHorseDre** [comment]
hey,
There's no config for it and the coordinates look hard coded in the diablo.js script.

I think you'd have to add the coordinates for the wp to the entrance to get this to work, but could be wrong.

`// path coordinates`
`    this.entranceToStar = [7794, 5517, 7791, 5491, 7768, 5459, 7775, 5424, 7817, 5458, 7777, 5408, 7769, 5379, 7777, 5357, 7809, 5359, 7805, 5330, 7780, 5317, 7791, 5293];
    this.starToVizA = [7759, 5295, 7734, 5295, 7716, 5295, 7718, 5276, 7697, 5292, 7678, 5293, 7665, 5276, 7662, 5314];
    this.starToVizB = [7759, 5295, 7734, 5295, 7716, 5295, 7701, 5315, 7666, 5313, 7653, 5284];
    this.starToSeisA = [7781, 5259, 7805, 5258, 7802, 5237, 7776, 5228, 7775, 5205, 7804, 5193, 7814, 5169, 7788, 5153];
    this.starToSeisB = [7781, 5259, 7805, 5258, 7802, 5237, 7776, 5228, 7811, 5218, 7807, 5194, 7779, 5193, 7774, 5160, 7803, 5154];
    this.starToInfA = [7809, 5268, 7834, 5306, 7852, 5280, 7852, 5310, 7869, 5294, 7895, 5295, 7919, 5290];
    this.starToInfB = [7809, 5268, 7834, 5306, 7852, 5280, 7852, 5310, 7869, 5294, 7895, 5274, 7927, 5275, 7932, 5297, 7923, 5313];
..
..
    if (Config.Diablo.Entrance) {`
`        Attack.clear(30, 0, false, this.sort);`
`        Pather.moveTo(7790, 5544);`

**Comment by gtoilet** [comment]
Shame blizzhachackers isn't still up I remember a post and solution there yrs ago

**Comment by Birk14** [comment]
Definitly a bummer, as I can set it to do a clear all of river but then it wont seem to run the chaos script and hit seals ect.

**Comment by DarkHorseDre** [comment]
yeah, cos you can do 'clearanyarea' for river of flame, but that will clear the whole thing, then you can do cs, but thats a loooong run.
the new limedrop branch does a diablo walk but only from entrance I believe..

afaik if you want to clear from wp to entrance, you need to tell the diablo script the path.. or maybe you just need to tell it the wp start coord and the entrance end coord as in 
`Pather.moveTo(7790, 5544);` ?  you should give it a try!   I just wish I understood the coordinates :(

**Comment by GZer** [comment]
Coords are based on exact map positioning.
you can find the coords of any spot in a map with

while(true){say(me.x+","+me.y);delay(2000);}

Honestly just add/modify the moveTo() or journeyTo() function calls in the Chaos.js script to Pather.moveToExit(108, true, true);

The bot will clear a Straight "ish" line (depends on your clearrange) to Chaos

**Comment by DarkHorseDre** [comment]
Thanks GZer - I may actually do this now I have a means of working out the positions (my previous post was about knowing how the coords are derived so I could predict - but I can just record them as I move too).

I'm guessing that if I provide a list of adjacent coords along the various paths leading to the entrance it will clear it all.

**question:**
what does journeyTo() and moveTo() do?
I know they are heavily used but not looked at them.   They both move, but does move only walk and journey tele?  is there any clearing if blocked for either?

**Comment by GZer** [comment]
journeyTo() is a high level function that calls moveTo() and many other lower functions like moveToExit().
You can investigate it under bots/common/Pather.js and I'd prefer if you learnt it yourself as kolton has done a very good job with code clarity.

You're better off using moveToExit(108,true,true); and set your Config.clearPath=40; and Config.clearType=0; that will clear pretty much everything along the way to the Chaos Sanctuary

FYI teleport is controlled by Pather.teleport=false; (or true;) which is a "public" variable within the Pather.js function

TLDR; Pather.teleport=true; means use Teleport. use journeyTo()/moveToExit() unless the layout is ALWAYS EXACTLY the same (Throne of Destruction / Chaos Sanctuary / Most Towns) then find the exact co-ordinates and use moveTo().

**Comment by DarkHorseDre** [comment]
Thanks man - I intend to look into it as previously posted, but a couple lines from you saves a lot of time working out multiple functions :)

is clearpath a config item not listed in the config files but supported in the code? (like `config.lowgold` etc).  I can't access my files to grep them at work - what does it do?  I currently have cleartype set to 0 as I clear quite a few places so it will work well for pre-cs entrance.

i dont mind teleport being used as long as it will clear along the way (as opposed to when it journeys to a boss or location and doesnt kill along the way - this was my query). sounds like i need to use journeyTo() with multiple coords in order to move along the varying map segments of that maze before the entrance.

**Comment by GZer** [comment]
Yes, there is a bots/common/config.js file with all the properties that you configure in charactername.js.
If you teleport it will only clear monsters within 40 radius of where you teleport whereas walking will clear anything within 40 radius of the walk path.

journeyTo() does not have a clearPath setting, search Pather.js for any function with clearPath to learn which ones are better suited to clearing.

**Comment by gbgodoy** [comment]
Config.ClearPath = { Areas: [107,108], Range: 18, Spectype: 0}			//chaos patch
